### ISSUE TITLE
Support for VirtualBox 5.2

### ISSUE THREAD

**Issue by BobNowadly** [hashicorp/vagrant#9090]
Please note that the Vagrant issue tracker is reserved for bug reports and
enhancements. For general usage questions, please use the Vagrant mailing list:
https://groups.google.com/forum/#!forum/vagrant-up. Thank you!

### Vagrant version
Vagrant version 2.0.0

### Host operating system
Windows 10 version 10.0.16299
VirtualBox 5.2

### Guest operating system
Ubuntu 16.04

### Debug output
When running vagrant up I get the following output 

The provider 'virtualbox' that was requested to back the machine
'dev' is reporting that it isn't usable on this system. The
reason is shown below:

Vagrant has detected that you have a version of VirtualBox installed
that is not supported by this version of Vagrant. Please install one of
the supported versions listed below to use Vagrant:

4.0, 4.1, 4.2, 4.3, 5.0, 5.1

A Vagrant update may also be available that adds support for the version
you specified. Please check www.vagrantup.com/downloads.html to download
the latest version.

**Comment by jenhsun** [comment]
My temporary solution can be found at https://github.com/hashicorp/vagrant/issues/7573

Here is what I did and it works.

Where: /opt/vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox
All changed need to be made with sudo.

Changed file: plugin.rb Add the following after Line 59:
autoload :Version_5_2, File.expand_path("../driver/version_5_2", __FILE__)

Changed file: driver/meta.rb Add the following after Line 64:
"5.2" => Version_5_2,

Copy file driver/version_5_1.rb to driver/version_5_2.rb and change two occurences of _5_0 to _5_1 and _5_1 to _5_2

**Comment by thelfensdrfer** [comment]
`autoload :Version_5_2, File.expand_path("../driver/version_5_2", FILE)` has to be `autoload :Version_5_2, File.expand_path("../driver/version_5_2", __FILE__)` i think?

**Comment by jenhsun** [comment]
@thelfensdrfer that's right, my typing error. Thanks and Updated.

**Comment by whiggs** [comment]
Followed the same instructions to get vagrant to work with new virtual box on Windows,  Got below error.  Going to make some changes to the indicated problem file and post back here what I did if I can fix it.  Or if I can't maybe someone else has some insight....
:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/driver/meta.rb:54: syntax error, unexpected =>, expecting keyword_end (SyntaxError)
                        "5.2" => Version_5_2,
                                ^
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/driver/meta.rb:54: dynamic constant assignment
C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/driver/meta.rb:187: syntax error, unexpected end-of-input, expecting keyword_end
        from C:/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/provider.rb:20:in `usable?'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/vagrantfile.rb:138:in `machine_config'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/vagrantfile.rb:45:in `machine'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/environment.rb:733:in `machine'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/plugin/v2/command.rb:177:in `block in with_target_vms'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/plugin/v2/command.rb:201:in `block in with_target_vms'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/plugin/v2/command.rb:183:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/plugin/v2/command.rb:183:in `with_target_vms'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/commands/up/command.rb:87:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/cli.rb:42:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/lib/vagrant/environment.rb:308:in `cli'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/bin/vagrant:138:in `<main>'

**Comment by icflorescu** [comment]
Although the issue specifically refers to Windows hosts, @jenhsun's temporary quick fix is **platform-independent**.
Just used it on a Linux Mint 18.2 / Ubuntu 16.04 host and worked like a charm.
Thanks!

**Comment by creacog** [comment]
To confirm that @jenhsun comment https://github.com/hashicorp/vagrant/issues/9090#issuecomment-338084000 also works on Mac OS X 10.11.6 (15G1611)

**Comment by redknitin** [comment]
There might be a fix for the issue in this commit: 7d73af5637de41f1e53b8f1ef2ea9baf76842dfb

**Comment by briancain** [comment]
Hey everyone: As I mentioned in a few other recent issues, support for VirtualBox 5.2 has already been merged into the master branch of Vagrant. We just haven't had a chance to roll out the next release. We're shooting for getting the next release out early next week. I'll go ahead and leave this issue open for now...Thanks!

**Comment by icflorescu** [comment]
@briancain - Thanks for letting us know, and good idea to leave this open. The quick fix above will help other people to work around the issue until next week.

**Comment by gutierri** [comment]
In archlinux the AUR [vagrant-git](https://aur.archlinux.org/packages/vagrant-git) package already solves this. Build from source :)

**Comment by deviouspk** [comment]
fix works on archlinux thanks

**Comment by sandric** [comment]
Can confirm that @jenhsun solution works on Windows 7.

**Comment by reneroboter** [comment]
@jenhsun save my day ^^

**Comment by i90rr** [comment]
Hello @briancain, what about adding an option to force the use of a specific version of the provider (
when Vagrant itself can't validate it), under the entire responsibility of the user (of course!)?

I'm not sure what would be the interface changes in this new release of VirtualBox that may affect Vagrant, but I guess that most of the times, stuff should keep working (kind of)-- and if not, well, that's why the provider isn't validated.

**Comment by adriandorin** [comment]
This fix works also for MacOS Sierra 10.12.6.
Cheers.

**Comment by nCrazed** [comment]
@gutierri have you noticed any noticeable changes in performance since virtualbox5.2 + vagrant-git?

**Comment by monotonee** [comment]
> Is it mandatory to hardcode the VirtualBox versions?

@ElMatella, my thoughts exactly. I strongly suspect that there is a better way. Both the VirtualBox 5.1 and 5.2 drivers merely extend the 5.0 driver, only adding a logger with (another) hard-coded VirtualBox version in the log name.

Perhaps implementing a driver factory to create the appropriate driver class for the detected provider version would be a more maintainable solution. The addition of a provider version parameter in the driver class' constructor would also allow the logger name to be dynamically generated. Only those provider versions that actually require additional driver functionality would need a discrete class.

**Comment by steelysteel** [comment]
Thanks @jenhsun for the work around until commit 7d73af5 is pushed out.

**Comment by glookie1** [comment]
is there already a plan when a new release with support for VirtualBox 5.2 will be released?

**Comment by biolypl** [comment]
@glookie1 It's ready, update your vagrant to 2.0.0 :+1:

**Comment by rphillips** [comment]
@biolypl The patch was created on September 8, and the tag for 2.0.0 created on September 7th. There will need to be a new release to suppose VirtualBox 5.2.

**Comment by gutierri** [comment]
No, it's the same thing, at least for me.

**Comment by sailorfred** [comment]
Wasn't 2.0.0 already available a few days ago?

Was a fix released under the same version number?

I ask because we were going through compatibility hell with bento/ubuntu-16.04, VirtualBox 5.2.30, and Vagrant 2.0.0 yesterday, so I don't see how 2.0.0 could be a 3 hour old release.

**Comment by sailorfred** [comment]
The 2.0.0 on your download is still broken:
`-----> Starting Kitchen (v1.16.0)
-----> Creating <default-ubuntu-1604>...
       The provider 'virtualbox' that was requested to back the machine
       'default' is reporting that it isn't usable on this system. The
       reason is shown below:
       
       Vagrant has detected that you have a version of VirtualBox installed
       that is not supported by this version of Vagrant. Please install one of
       the supported versions listed below to use Vagrant:
       
       4.0, 4.1, 4.2, 4.3, 5.0, 5.1
       
       A Vagrant update may also be available that adds support for the version
       you specified. Please check www.vagrantup.com/downloads.html to download
       the latest version.
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [Expected process to exit with [0], but received '1'
---- Begin output of vagrant up --no-provision --provider virtualbox ----
STDOUT: 
STDERR: The provider 'virtualbox' that was requested to back the machine
'default' is reporting that it isn't usable on this system. The
reason is shown below:

Vagrant has detected that you have a version of VirtualBox installed
that is not supported by this version of Vagrant. Please install one of
the supported versions listed below to use Vagrant:

4.0, 4.1, 4.2, 4.3, 5.0, 5.1

A Vagrant update may also be available that adds support for the version
you specified. Please check www.vagrantup.com/downloads.html to download
the latest version.
---- End output of vagrant up --no-provision --provider virtualbox ----
Ran vagrant up --no-provision --provider virtualbox returned 1] on default-ubuntu-1604
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

make: *** [cb_smoke_test_1604_authentic8] Error 1
freds-mbp:chef fcox$ vagrant --version
Vagrant 2.0.0
`

**Comment by briancain** [comment]
@sailorfred - You are likely getting that message because you have VirtualBox 5.2 installed on your system. I recommend downgrading to the latest VirtualBox 5.1.x, and you should be able to get Vagrant working again.

There's a few more tasks for us to finish, and the next release that includes support for VirtualBox 5.2 should be out soon! Thanks for your patience everyone.

**Comment by sailorfred** [comment]
It was the @biolypl response that confused me:

> @glookie1 It's ready, update your vagrant to 2.0.0 :+1:

**Comment by DonRichards** [comment]
It's never good when someone suggest downgrading to use another product. But thanks for the work!

**Comment by ronanmccoy** [comment]
Just adding my $0.02: on macOS Sierra, using Virtualbox 5.2.0 and Vagrant 2.0.0, and the solution offered by @jenhsun worked for me.

**Comment by wunderdojo** [comment]
Ubuntu 17.10 is working with VirtualBox 5.1 but does work with 5.2. There file that @jenhsun says to change doesn't exist in /opt on 17.10 and I'm baffled as to where it is. Tried Vagrant 1.9.1 and Vagrant 2.0. Both install fine but aren't getting installed in some other location. Any ideas?

**Comment by KyleGoddard** [comment]
Mac OS X 10.12.6
VirtualBox 5.2.0
Vagrant 2.0.0

Jenshun's fix worked for me.

**Comment by lihaibh** [comment]
How to fix on windows?

**Comment by DonRichards** [comment]
@ronanmccoy That's great! Thanks for pointing that out. ⭐️

**Comment by nvdbeek** [comment]
@briancain To bad that you say previous week "We're shooting for getting the next release out early next week." but still no fix. Is there some information when the next release is available?

**Comment by voodookoop** [comment]
For configuration: it is enough to add `"5.2" => Version_5_1,` to **meta.rb** after row 64. No other changes required to make it work.

**Comment by zaro** [comment]
@voodookoop  solution worked perfectly for me on macOS Sierra.

**Comment by chrisroberts** [comment]
@nvdbeek I'm getting some final updates wrapped up within the installers. We were indeed shooting for a release early this week but have had some other things to deal with as well. Once the installers work is complete the releases will be ready.

Cheers!

**Comment by lihaibh** [comment]
@jenhsun i get an error when changing those files in windows.
```
C:/Program Files (x86)/Develop/Env/HashiCorp/Vagrant/embedded/mingw64/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- C:/Program Files (x86)/Develop/Env/HashiCorp/Vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/driver/version_5_2 (LoadError)
```

**Comment by mnpenner** [comment]
@lihaibh Exactly [the same way](https://github.com/hashicorp/vagrant/issues/9090#issuecomment-338084000). Files are located at `C:\HashiCorp\Vagrant\embedded\gems\gems\vagrant-2.0.0\plugins\providers\virtualbox`.

**Comment by ericleu** [comment]
vagrant in homebrew is still broken (at least not picking up the latest push). @voodookoop 's fix works for me:

> For configuration: it is enough to add "5.2" => Version_5_1, to meta.rb after row 64. No other changes required to make it work.

**Comment by jaraco** [comment]
If you're on macOS with Vagrant installed from the website, the following command will patch your installation as recommended by @voodookoop:

```
$ sudo sed -i '/"5.1" => Version_5_1,/a\
\            "5.2" => Version_5_1,
' /opt/vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/driver/meta.rb
```

Note there is are two carriage returns in that command, one after the `/a\` and another after `_5_1,`, but if you copy/paste that command to terminal, it should do the trick.

**Comment by scttfrdmn** [comment]
@jaraco thanks - I needed to modify your command slightly on OSX 10.12.6

`sudo sed -i '' -e  '/"5.1" => Version_5_1,/a\
\            "5.2" => Version_5_1,
' /opt/vagrant/embedded/gems/gems/vagrant-2.0.0/plugins/providers/virtualbox/driver/meta.rb`

**Comment by soliverprofesional** [comment]
Same issue/solution with Fedora26. 
Thanks guys, good work.

**Comment by zuernBernhard** [comment]
the solution from @voodookoop also works for me on mac os 10.13 (high sierra)

**Comment by joelhandwell** [comment]
Now vagrant 2.0.1 is out. Those who want to solve this can upgrade vagrant.

**Comment by e-kolpakov** [comment]
@joelhandwell Not on the [download page](https://www.vagrantup.com/downloads.html) yet - is there some  other place?

**Comment by joelhandwell** [comment]
@e-kolpakov sorry not yet as binary or installer. Only I found https://github.com/hashicorp/vagrant/releases/tag/v2.0.1

**Comment by joelhandwell** [comment]
And it's out now as binary installer at download page.

**Comment by SummerMagic** [comment]
update vagrant success

**Comment by pmoranga** [comment]
Indeed works with 2.0.1, needs to update documentation at https://www.vagrantup.com/docs/virtualbox/ to mention it.

**Comment by briancain** [comment]
@pmoranga - Good catch, I'm opening a pull request now to fix that. Thanks!

**Comment by briancain** [comment]
Also closing this, as it has been fixed and released!

**Comment by azzzdiggr** [comment]
I am getting the issue with all version which I had used 
if some could help me with that.
VMware tool is much effective than this  Vagrant in terms of compatibility.
I thought its quite good using docker kind of process to establish machine (BOX)
but it sucks 👎

**Comment by chrisroberts** [comment]
@azzzdiggr If you are receiving an error about VirtualBox 5.2 not being supported than you likely need to uninstall Vagrant and install it again using an official release package from the downloads page: https://www.vagrantup.com/downloads.html

Otherwise, please feel free to create a new issue and provide the requested information in the template.

Cheers!

**Comment by michaelrodas** [comment]
@jenhsun Thank you so much, it really helped me
